pkg/types/validation/installconfig.go

```diff
@@ -131,6 +133,15 @@ func ValidateInstallConfig(c *types.InstallConfig, usingAgentMethod bool) field.
 	if _, ok := validPublishingStrategies[c.Publish]; !ok {
 		allErrs = append(allErrs, field.NotSupported(field.NewPath("publish"), c.Publish, validPublishingStrategyValues))
 	}
+	allErrs = append(allErrs, validateImageDigestSources(c.ImageDigestSources, field.NewPath("imageDigestSources"))...)
+	if _, ok := validPublishingStrategies[c.Publish]; !ok {
+		allErrs = append(allErrs, field.NotSupported(field.NewPath("publish"), c.Publish, validPublishingStrategyValues))
+	}
+	if len(c.ImageContentSources) != 0 && len(c.ImageDigestSources) != 0 {
```

Comment:
Minor point, but I'd nice to have all the validations related to the ImageDigestSource contained within one function.

Reply:
Keep it as is. From the review https://github.com/openshift/installer/pull/6235#discussion_r1150947827 we only have to do the validation once in installconfig.go. 
I removed duplicates of this code from other files.